conctr.agent.class.nut

```diff
@@ -164,23 +195,78 @@ class Conctr {
                 local device_result = { "ids": ids, "body": body, "error": error};
                 device.send(DATA_EVENT, device_result);
             } else if (error != null) {
-                server.error("Conctr: " + error);
+                server.error("Conctr Error: " + error);
             }
 
         }.bindenv(this));
     }
+
     /**
-     * Returns a table containing the last recieved location update from the device
-     * @return {Table} last known location with keys location and ts (timestamp) 
-     * {
-     *     location,
-     *     ts
-     * }
+     * Sends a request to the device to send its current location (array of wifis) if conditions in current location sending opts are met. 
+     * Note: devcie will send through using its internal sendData function, we will not wait send location within the current payload.
+     * 
      */
-    function getLastKnownLocation(){
-        return lastKnownLoc;
+    function _getLocation(){
+
+        if(_DEBUG){
+            server.log("Conctr: Checking whether to retrieve location.");
+        }
+
+         if (!_locationRecording) {
+
+            if(_DEBUG){
+                server.log("Conctr: Location recording is not enabled");
+            }
+            // not recording location 
+            return;
+
+        } else {
+            server.log("curr loc timeout "+_locationTimeout);
+            server.log("unix timestamp "+_getUnixMS());
+            server.log("_interval "+_interval);
+           
+            //check new location scan conditions are met and search for proximal wifi networks
+            if ((_sendLocationOnce == true && _locationSent == false) || ((_sendLocationOnce == false) && (_locationRecording == true) && (_locationTimeout < _getUnixMS()))) {
+                
+                if(_DEBUG){
+                    server.log("Conctr: Retrieving location from device");
+                }
+
+                //update timeout 
+                _locationTimeout = _getUnixMS() + _interval;
+
+                //update flagg to show we sent location.
+                _locationSent = true;
+
+                //Request location from device
+                device.send(LOCATION_REQ,"");
+
+            } else {
+                //conditions for new location search (using wifi networks) not met
+                return;
+            }
+        }
+    }
+
+    function _getUnixMS(){
+
+        local ts = date();
+
+        return format("%d%03d", ts.time, ts.usec/1000).tointeger();
     }
 
+    function _setLocationOpts(opts = {}){
```

Comment:
Missing function documentation.
Should this be called `setOpts` like on the device?